File: BioMetricAuthJava/app/src/main/java/com/example/biometricauthjava/PinEntryActivity.java
package com.example.biometricauthjava;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class PinEntryActivity extends AppCompatActivity {

    private EditText etPin;
    private Button btnSubmitPin;
    private PinStorageManager pinStorage;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_pin_entry);

        etPin = findViewById(R.id.etPin);
        btnSubmitPin = findViewById(R.id.btnSubmitPin);
        pinStorage = new PinStorageManager(this);

        btnSubmitPin.setOnClickListener(view -> {
            String enteredPin = etPin.getText().toString();
            String savedPin = pinStorage.getPin();

            if (enteredPin.equals(savedPin)) {
                Toast.makeText(this, "PIN correct", Toast.LENGTH_SHORT).show();
                startActivity(new Intent(PinEntryActivity.this, HomeActivity.class));
                finish();
            } else {
                Toast.makeText(this, "Incorrect PIN", Toast.LENGTH_SHORT).show();
            }
        });
    }

    @Override
    public void onBackPressed() {
        super.onBackPressed();
        startActivity(new Intent(this, MainActivity.class));
        finish();
    }
}

File: BioMetricAuthKotlin/app/src/main/java/com/example/biometricauthkotlin/PinEntryActivity.kt
package com.example.biometricauthkotlin

import android.content.Intent
import android.os.Bundle
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.example.biometricauthkotlin.databinding.ActivityPinEntryBinding
import kotlin.jvm.java

class PinEntryActivity : AppCompatActivity() {
    private lateinit var binding : ActivityPinEntryBinding

    private lateinit var pinStorage: PinStorageManager

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        binding = ActivityPinEntryBinding.inflate(layoutInflater)
        setContentView(binding.root)

        pinStorage = PinStorageManager(this)

        binding.btnSubmitPin.setOnClickListener {
            val enteredPin = binding.etPin.text.toString()
            val correctPin = pinStorage.getPin()

            if (enteredPin == correctPin) {
                Toast.makeText(this, "PIN correct", Toast.LENGTH_SHORT).show()
                startActivity(Intent(this, HomeActivity::class.java))
                finish()
            } else {
                Toast.makeText(this, "Incorrect PIN", Toast.LENGTH_SHORT).show()
            }
        }

    }

    override fun onBackPressed() {
        super.onBackPressed()
        startActivity(Intent(this, MainActivity::class.java))
        finish()
    }
}